core/optaplanner-core-impl/src/main/java/org/optaplanner/core/config/heuristic/selector/move/generic/RuinMoveSelectorConfig.java

```diff
@@ -0,0 +1,82 @@
+package org.optaplanner.core.config.heuristic.selector.move.generic;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import org.optaplanner.core.config.heuristic.selector.entity.EntitySelectorConfig;
+import org.optaplanner.core.config.heuristic.selector.move.MoveSelectorConfig;
+import org.optaplanner.core.config.util.ConfigUtils;
+
+public class RuinMoveSelectorConfig extends MoveSelectorConfig<RuinMoveSelectorConfig> {
```

Comment:
At first, I was going to ask why does this exist, if there already is a `RuinPhaseConfig` with the same properties. Then I realized this is because `MoveSelectorFactory` actually requires a `MoveSelectorConfig`, so the phase config needs to create one.

Although this is completely logical, it poses a problem. Everything in the `config` package is considered API, and it will be recommended to people in their IDE. So people will be using this class, not knowing or understanding that they should be using `RuinPhaseConfig` instead. Ideally, this class would be package-private, but then that would make it inaccessible to the `MoveSelectorFactory`.

I'm not exactly sure of the best way to solve it, but I know I don't want this type exposed publicly.

Reply:
[4e2f380](https://github.com/kiegroup/optaplanner/pull/2752/commits/4e2f38037b8ad3720a0295ca0921ae720d04d5cd), let me know what you think

Reply:
Hence we loose this AbstractMoveSelectorFactory.buildMoveSelector logic (caching, shuffling, sorting, filtering...), but I wonder whether we even need it for ruin move